app/lib/actions/katello/content_view_version/create_export_history.rb

```diff
@@ -0,0 +1,67 @@
+module Actions
+  module Katello
+    module ContentViewVersion
+      class CreateExportHistory < Actions::EntryAction
+        input_format do
+          param :smart_proxy_id, Integer
+          param :exporter_data, Hash
+          param :content_view_version_id, Integer
+          param :from_content_view_version_id, Integer
+          param :destination_server, String
+        end
+
+        output_format do
+          param :export_history_id, Integer
+          param :path, String
+        end
+
+        def run
+          smart_proxy = ::SmartProxy.find(input[:smart_proxy_id])
+          api = ::Katello::Pulp3::Api::Core.new(smart_proxy)
+          Rails.logger.info input[:exporter_data][:pulp_href]
+          export_data = api.export_api.list(input[:exporter_data][:pulp_href]).results.first
+          output[:exported_file_checksum] = export_data.output_file_info
+          file_name = output[:exported_file_checksum].first&.first
+          path = File.dirname(file_name.to_s)
+          output[:export_path] = path
+          cvv = ::Katello::ContentViewVersion.find(input[:content_view_version_id])
+          from_cvv = ::Katello::ContentViewVersion.find(input[:from_content_view_version_id]) unless input[:from_content_view_version_id].blank?
+
+          export_metadata = ::Katello::Pulp3::ContentViewVersion::Export.new(
+                                                     content_view_version: cvv,
+                                                     smart_proxy: smart_proxy,
+                                                     from_content_view_version: from_cvv).generate_metadata
+
+          toc_path_info = output[:exported_file_checksum].find { |item| item.first.end_with?("toc.json") }
+          export_metadata[:toc] = File.basename(toc_path_info.first)
+
+          history = ::Katello::ContentViewVersionExportHistory.new(
+            content_view_version_id: input[:content_view_version_id],
+            destination_server: input[:destination_server],
+            path: path,
+            metadata: export_metadata
+          )
+          history.export_type = history.export_type_from_metadata
```

Comment:
Why do you need to do this ?? shouldn't https://github.com/Katello/katello/blob/master/app/models/katello/content_view_version_export_history.rb#L22 already do that ?

Reply:
I dont think its necessary try removing and see if it sets the wrong thing. If you Have to do it 
```suggestion
          history.set_export_type
```

Reply:
I did it because `generate_audit_comment` uses `export_type`:
```rb
"#{export_history.export_type&.capitalize} #{export_descriptor} created by #{user_full_name}"
```
However, I need to `generate_audit_comment` before the record is saved, so the `before_validation :set_export_type` won't have run yet.  I did consider doing `history.validate`, but left it like this so it was more clear that using this value was the reason.

Reply:
What do you think about renaming `set_export_type` to `update_export_type_from_metadata!`  instead and calling that. I just don't like the world outside that model to know that export_type is not set and its value is based of  `export_type_from_metadata`

Reply:
Looks like adding the `after_initialize` callback in the model saves you from having to call it here! :+1: